src/holo-crypto/crates/holo-crypto/src/lib.rs

```diff
@@ -0,0 +1,85 @@
+pub mod error;
+
+use argon2min::Argon2;
+use base64::encode;
+use ed25519_dalek::{Keypair as EdKeypair, SecretKey, SECRET_KEY_LENGTH};
+pub use error::HoloCryptoError;
+
+pub struct KeyPair(EdKeypair);
+
+const ARGON2_ADDITIONAL_DATA: &[u8] = b"holocrypto";
+const ARGON2_MAX_PASSPHRASE_LEN: usize = (2 ^ 32) - 1;
+const ARGON2_MAX_SALT_LEN: usize = (2 ^ 32) - 1;
+const ARGON2_MIN_SALT_LEN: usize = 8;
+
+impl KeyPair {
+    /// Create a new keypair given a passphrase and salt
+    /// Passphrase must be between 0 and 2^32-1 bytes
+    /// Salt must be between 8 and 2^32 - 1 bytes
+    pub fn new(passphrase: &str, salt: &str) -> Result<Self, HoloCryptoError> {
+        let passphrase_bytes = passphrase.as_bytes();
+        let salt_bytes = salt.as_bytes();
+        if passphrase_bytes.len() > ARGON2_MAX_PASSPHRASE_LEN {
+            return Err(HoloCryptoError::Custom(
+                "Passphrase exceeds maximum allowable number of bytes".to_string(),
+            ));
+        }
+        match salt_bytes.len() {
+            0..=ARGON2_MAX_SALT_LEN if salt_bytes.len() < ARGON2_MAX_SALT_LEN => Err(
+                HoloCryptoError::Custom("Salt must be at least 8 bytes".to_string()),
+            ),
+            ARGON2_MIN_SALT_LEN..=ARGON2_MAX_SALT_LEN => Ok(()),
+            _ => Err(HoloCryptoError::Custom(
+                "Salt exceeds maximum allowable number of bytes".to_string(),
+            )),
+        }?;
+
+        // generate a 32 byte password hash using the Argon2 algo
+        let mut secretkey_bytes = [0; SECRET_KEY_LENGTH];
+
+        //The number of block matrix iterations to perform. Increasing
+        // this forces hashing to take longer. Must be between 1 and 2^32 - 1.
+        let passes = 2;
```

Comment:
TBH, I would inline all of that into `Argon2::new` call. Or at the very least, define as constants.